lib/grape/api/instance.rb

```diff
@@ -79,9 +79,13 @@ def cascade(value = nil)
           end
         end
 
+        def compile!
+          LOCK.synchronize { compile } unless instance
```

Comment:
Shouldn't you be checking for instance inside the synchronize?

```ruby
def compile!
    return if instance
    LOCK.syncrhonize { compile unless instance } 
 end
```

Reply:
that does make sense, even though it's not what was happening on the original method